From 7e88b27dfe84f637e1dcab4784dde62c804fc55c Mon Sep 17 00:00:00 2001
From: David Allsopp <david.allsopp@metastack.com>
Date: Thu, 5 Dec 2019 12:11:51 +0000
Subject: [PATCH] Backport PR5453 to 3.12.1

---
 configure | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 47cc203512..7b1fd1dcf6 100755
--- a/configure
+++ b/configure
@@ -1357,7 +1357,10 @@ for dir in \
     /lib/usr/lib/X11      \
                           \
     /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+    /usr/openwin/share/lib    \
+                              \
+    /usr/lib/i386-linux-gnu   \
+    /usr/lib/x86_64-linux-gnu \
     ; \
 do
   if test -f $dir/libX11.a || \
-- 
2.20.1

